src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -191,6 +192,16 @@ export interface DiscoverGridProps {
    * Document detail view component
    */
   DocumentView?: typeof DiscoverGridFlyout;
+  /**
+   * Service dependencies
+   */
+  services: {
+    fieldFormats: FieldFormatsStart;
+    addBasePath: DiscoverServices['addBasePath'];
+    uiSettings: DiscoverServices['uiSettings'];
+    dataViewFieldEditor: DiscoverServices['dataViewFieldEditor'];
+    toastNotifications: DiscoverServices['toastNotifications'];
```

Comment:
To further decouple the grid, it would be better to use the original types, for example `toastNotifications: ToastsStart`